### ISSUE TITLE
Please fix security issues.

### ISSUE THREAD

**Issue by Mister-Hope** [commitizen/cz-cli#914]
![image](https://user-images.githubusercontent.com/33315834/159410766-96c28fe4-2365-4b91-ba9e-be17f5f374f7.png)
![image](https://user-images.githubusercontent.com/33315834/159410788-f95b7d8f-df85-436a-a8a1-e265762dd370.png)

**Comment by Mister-Hope** [comment]
Also, is there any strong reson why commitizen choose to lock minimist version? No other packages in my project is doing that.

**Comment by kilahm** [comment]
See #913 - looks like the auto-merge of the dependency update is waiting on the stabilityDays setting of RenovateBot.  I tried to find the value, but I was unable to trace it beyond https://github.com/commitizen/commitizen-renovate-config/blob/master/default.json

**Comment by Mister-Hope** [comment]
I am just curious why this dep's version is fixed. If we do not pin it with an exact version, this can be fixed by updating deps myself.

**Comment by Mister-Hope** [comment]
any update about this one?

**Comment by LinusU** [comment]
I very much prefer non-pinned dependencies myself, but it was made this way before I started contributing. These days I'm not personally using this tool, and I believe that the publishing workflow doesn't work anymore (#897). Because of this I don't have much bandwidth to fix things here...

If it sounds like a good way forward, I would be open to merging a pull request that changes the dependencies to be specified with `^` (it would need to update the renovate config as well), and then doing a manual release from my computer 🚀

**Comment by urecio** [comment]
Hi, any update on this? Looks like only publishing is needed now

**Comment by zSakuraEvilz** [comment]
Hi, any update on this? I need to use it but still have security issues. So my company does not allow the use.

**Comment by Mister-Hope** [comment]
Half a month bro 🤡, any schedule? Or shall I fork and publish a temp version first?

**Comment by Mister-Hope** [comment]
Just noticed that the Severity is lifted to Critical. And I am sure that this will bother many people due to their company policity.

**Comment by Mister-Hope** [comment]
@jimthedev Help needed here

**Comment by techmunk** [comment]
We worked around this using the `overrides` package.json property added in npm v8. Something like the below:
```json
  "overrides": {
    "commitizen": {
      "inquirer": "8.2.0",
      "minimist": "1.2.6"
    }
  }
  ```
  
  Not ideal, but might help some people move forward. I had to remove node modules and possibly package-lock.json as well before doing npm install to get the changes to take effect.

Commitizen seems to be working fine with these changes, and npm audit is happy now too.

**Comment by caleb-mabry** [comment]
If the above solution doesn't work for you, we used https://www.npmjs.com/package/npm-force-resolutions to resolve the vulnerability for the time being.

**Comment by noahnu** [comment]
If using Yarn modern, you can set this resolution in your project's package.json (equivalent of npm overrides):

```
"minimist@1.2.5": "npm:1.2.6"
```

**Comment by enaukkarinen** [comment]
Any update on this? I don't see a workaround for this for people using NPM 7 :(

**Comment by hasghari** [comment]
The `master` branch has this fix. Is there going to be a release soon?

**Comment by yvesnrb** [comment]
Anyone know of an alternative program for generating git commits following the cz-conventional-changelog format?

**Comment by techmunk** [comment]
@yvesnrb best I know of is https://commitizen-tools.github.io/commitizen/, which uses python.

**Comment by machadodev** [comment]
Almost a month now. Still no release date?

**Comment by pumano** [comment]
@jimthedev please help with it. Many critical vulnerabilities here in package, need updated dependencies and release.

**Comment by spock123** [comment]
This project appears dead.  Need to find alternatives

**Comment by davidh-unmind** [comment]
can we get a release please maintainers?

**Comment by ryansonshine** [comment]
The fix has already been merged into `master`, the problem is that it appears the maintainers [do not have access to publish][1] to npm, and have not had access for over a year now.

I couldn't wait any longer for this due to the number of tickets I've had opened because of this so I've created [a fork][2] with an automated publishing lifecycle, which I'm now using in all of my projects.

If anyone is looking for a solution that doesn't involve using overrides feel free to use it; it will continue to get security updates from this repository going forward since it'll be supporting enterprise projects.

### Migrate

Run the following and commit the result in your project to complete the migration.

npm

```sh
npm uninstall commitizen cz-conventional-changelog
npx @ryansonshine/commitizen init @ryansonshine/cz-conventional-changelog --includeCommitizen --force
```

yarn

```sh
npm uninstall commitizen cz-conventional-changelog
npx @ryansonshine/commitizen init @ryansonshine/cz-conventional-changelog --includeCommitizen --force --yarn
```


[1]:https://github.com/commitizen/cz-cli/issues/897
[2]:https://github.com/ryansonshine/cz-cli

**Comment by Mister-Hope** [comment]
> The fix has already been merged into `master`, the problem is that it appears the maintainers [do not have access to publish](https://github.com/commitizen/cz-cli/issues/897) to npm, and have not had access for over a year now.

I don’t think so, the 1st contributor was publishing release in the past years ➡️@jimthedev And he still has activity in the past week on GitHub. As the project creator and 1st contributor, I think he should be still in charged with this package, and I am not supposing he just “lost access”

I would like to prefer to believe that he dropped this package maintenance and prefer to deprecate it (but without any announcement)

This is open source, yet he should not be blamed, but I still think that his behavior does not fit the community environment and does have negative infect in the nook ecosystem. At least he fails other contributor who still working on this project.

**Comment by ryansonshine** [comment]
He may have activity within the past week but the last time the package was published was last year, there have been plenty of updates since then. See below:
![image](https://user-images.githubusercontent.com/9534477/169249897-2b53219c-b9a2-4dc3-9f16-8d7eb61847d1.png)

We're going on 60 days since this issue was opened so I took action for myself and the projects I'm involved in. Mine will continue to get security updates for my projects, if anyone else can benefit from that then great; if not well that's fine too :)

**Comment by Mister-Hope** [comment]
I am just expressing the following point:

1. He publish the first version of package, and he managed to switch to a bot account to continue publishing, I DO NOT believe he lost access. If he stop using github account or I only opened this issue for a week, I don't feel strange, but after 2 months with github activity, we can definitely tell that he choosed "ignore" from "pickup this package and give some active contributor rights to maintain".
2. He is acting as he quit this project (or stop maintaining this project), and I think that's pretty ok. But I think as an open source project with > 20 contributor, he should try to keep this project running and give rights to other active contibutor instead of failing them, yet we are still meeting no one publishing this package. I agree that opensource authors have rights to deal with their proejcts, as I have open projects too. But I think that it's necessary to keep basic polite to do either:
 
   - archieve the repo and make annoucement
   - give rights to active contibuotr to keep running it

   specially when this project as >10k stars and a high download number

**Comment by ryansonshine** [comment]
Ah my mistake, I misread your initial comment. 

I agree with both of your statements. I'll update my comment accordingly.

**Comment by Manokii** [comment]
> ```shell
> @ryansonshine/cz-conventional-changelog
> ```

yarn migration doesn't seem to work on a monorepo, wasn't able to submit an issue in the forked repo so im adding it here

**Comment by ryansonshine** [comment]
Thanks for the feedback @Manokii !

I've went ahead an enabled issues on the repository as well as updated the installation instructions for yarn on [my initial comment](https://github.com/commitizen/cz-cli/issues/914#issuecomment-1131383383).

**Comment by Zhengqbbb** [comment]
> @yvesnrb best I know of is https://commitizen-tools.github.io/commitizen/, which uses python.

Try run command `npx czg` in your any project

**Comment by kdmcguire** [comment]
I believe this is fixed in the new release, [4.2.5](https://github.com/commitizen/cz-cli/releases/tag/v4.2.5).
